exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/EndToEndTest.java

```diff
@@ -38,38 +36,46 @@
   private static final SpanId PARENT_SPAN_ID = new SpanId(54321);
   private static final String SPAN_NAME = "MySpanName";
   private static final long START_EPOCH_NANOS = TimeUnit.SECONDS.toNanos(3000) + 200;
-  private static final long START_SECONDS = TimeUnit.NANOSECONDS.toSeconds(START_EPOCH_NANOS);
-  private static final int START_NANOS = (int) (START_EPOCH_NANOS - TimeUnit.SECONDS.toNanos(START_SECONDS));
   private static final long END_EPOCH_NANOS = TimeUnit.SECONDS.toNanos(3001) + 255;
-  private static final long END_SECONDS = TimeUnit.NANOSECONDS.toSeconds(END_EPOCH_NANOS);
-  private static final int END_NANOS = (int) (END_EPOCH_NANOS - TimeUnit.SECONDS.toNanos(END_SECONDS));
-  private static final String SERVER_PREFIX = "Recv.";
-  private static final String AGENT_LABEL_KEY = "g.co/agent";
-  private static final String AGENT_LABEL_VALUE_STRING = "opentelemetry-java [0.6.0]";
-  private static final AttributeValue AGENT_LABEL_VALUE = AttributeValue.newBuilder()
-          .setStringValue(TruncatableString.newBuilder().setValue(AGENT_LABEL_VALUE_STRING).setTruncatedByteCount(0).build()).build();
-  private static final Span.Attributes ATTRIBUTES = Span.Attributes.newBuilder()
-          .setDroppedAttributesCount(0)
-          .putAttributeMap(AGENT_LABEL_KEY, AGENT_LABEL_VALUE)
-          .build();
-  private static final Span.TimeEvents TIME_EVENTS = Span.TimeEvents.newBuilder().build();
-  private static final io.opentelemetry.trace.Status SPAN_DATA_STATUS = io.opentelemetry.trace.Status.OK;
-  private static final Status SPAN_STATUS = Status.newBuilder().setCode(SPAN_DATA_STATUS.getCanonicalCode().value())
-          .build();
-  private static final Span.Links LINKS = Span.Links.newBuilder().setDroppedLinksCount(0).build();
-
-  @Mock
-  TraceServiceClient mockTraceServiceClient;
+  private static final Status SPAN_DATA_STATUS = Status.OK;
+  private static final String LOCALHOST = "127.0.0.1";
+
+  private MockCloudTraceClient mockCloudTraceClient;
   private TraceExporter exporter;
+  private Process mockServerProcess;
+
 
   @Before
-  public void setup() {
-    MockitoAnnotations.initMocks(this);
+  public void setup() throws IOException {
+    // Find a free port to spin up our server at.
+    ServerSocket socket = new ServerSocket(0);
+    int port = socket.getLocalPort();
+    String address = String.format("%s:%d", LOCALHOST, port);
+    socket.close();
+
+    // Start the mock server. This assumes the binary is present and in $PATH.
```

Comment:
So once we create the CircleCI build we will need to install the binary before executing this test, correct?

Reply:
Yes, what the Python exporter does is have a shell script `curl` the release binary from our repo, and then add it to `$PATH` (Python uses `Tox` to add the binary to `{envbindir}`, which is already part of `$PATH`, CircleCI probably needs to manually do this). The script then creates a symlink for the binary to avoid having to deal with versions. See [this PR](https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/pull/30) for the script).